### PR TITLE
7 add comments to users

### PR DESCRIPTION
**Pull request by eliotsnowdon**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -165,4 +165,38 @@ describe('5-get reviews by Id', () => {
       .expect(404)
     })
   })
-        
\ No newline at end of file
+describe('7-post', () => {
+  test('status:201 adds comments to path', () => {
```
Comment (lewispricey) [inline __tests__/app.test.js:169]: Really like this test, explicitly expecting the correct author and body but testing the types of the other properties!

File: controller.js [hunk 2]
```diff
@@ -40,3 +40,14 @@ exports.getComFromId = (req, res, next) => {
         next(err)
     })
 }
+
+exports.postComments = (req, res, next) => {
+    insertComment(req.body, req.params.review_id)
+    .then((comments) => {
+        res.status(201).send({comments})
```
Comment (lewispricey) [inline controller.js:47]: Great choice of key to send the data back on 👍

File: model.js [hunk 3]
```diff
@@ -34,4 +34,13 @@ exports.selectCommentById = (review_id) => {
     .then((result) => { 
          return result.rows
     })
-}
\ No newline at end of file
+}
+
+exports.insertComment = (newComment, Id) => {
+    const {username, body} = newComment
+    let queryStr = 'INSERT INTO comments (author, body, review_id) VALUES ($1, $2, $3) RETURNING *'
```
Comment (lewispricey) [inline model.js:41]: Nice use of parameterization to prevent SQL injection attacks.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -165,4 +165,38 @@ describe('5-get reviews by Id', () => {
...
+  })
+  test('adds comments to invalid username', () => {
+    newComment = {
+      username: "jambo",
+      body:"i am him"
+    }
+    return request(app)
+    .post('/api/reviews/3/comments')
+    .send(newComment)
+    .expect(400)
```
Comment (lewispricey) [inline __tests__/app.test.js:219]: For this test we'd expect a 404 (username is valid but does not exist)

**Review by lewispricey — CHANGES_REQUESTED** [review]
Great work so far Eliot.

Really like the tests you've written but there are a few more things that need testing:
- what happens if the user passes in an extra property (eg. votes: 10) is the extra property ignored as it should be?
- what happens when the article id is not a valid id (eg. 'hello')
- an id that doesn't exist in the database
- the comment the user passes in is missing a username or body

**Comment by lewispricey** [comment]
> 

Hint: PSQL is really helpful for inserting so be sure to console log error codes that you can handle in your errorhandling function

**Comment by Dutton1990** [comment]
Nice job Eliot! I think your test descriptions could be a little clearer but the code is all good. You can merge this!

**Pull request merged**